cockpit/src/subscriptions-client.js

```diff
@@ -124,161 +92,104 @@ function getSubscriptionDetails() {
-            } else if (message.indexOf('The system has been registered') !== -1) {
-                /*
-                 * Currently we don't separate registration & subscription.
-                 * Our auto-attach may have failed, so close the dialog and
-                 * update status.
-                 */
-                dfd.resolve();
-                return;
-            } else {
-                // unrecognized output
-                console.log("unrecognized subscription-manager failure output: ", ex, message);
+            else {
+                return registerService.call('Register', [subscriptionDetails.org, subscriptionDetails.user, subscriptionDetails.password, options, connection_options]);
             }
-            var error = new Error(message);
-            dfd.reject(error);
+        })
+        .catch(function(error) {
+            console.error('error initiating registration', error)
+            throw error;
+        })
+        .then(function() {
+            return registerServer.Stop();
+        })
+        .catch(function(error) {
+            console.error('error registering', error);
+            throw error;
+        })
+        .then(function() {
+            return attachService.AutoAttach('', {});
+        })
+        .catch(function(error) {
+            console.error('error stopping registration bus!', error);
+            throw error;
+        })
+        .then(function(result) {
+            console.log('requesting update');
+            requestUpdate();
+        })
+        .catch(function(error) {
+            console.error('error during attach', error);
+            throw error;
         });
-
-    promise = dfd.promise();
-    promise.cancel = function cancel() {
-        process.close("cancelled");
-        // we have no idea what the current state is
-        requestUpdate();
-    };
-
-    return promise;
 };
 
 client.unregisterSystem = function() {
-    var dfd = cockpit.defer();
-
-    var args = ['subscription-manager', 'unregister'];
-
-    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
-     * without translations and parse the output
-     */
-    var process = cockpit.spawn(args, {
-        directory: '/',
-        superuser: "require",
-        environ: ['LC_ALL=C'],
-        err: "out"
-    });
-
-    client.subscriptionStatus.status = "unregistering";
+    client.subscriptionStatus.status = "Unregistering";
     needRender();
-    var promise;
-    var buffer = '';
-    process
-        .input('')
-        .stream(function(text) {
-            buffer += text;
-        })
-        .done(function(output) {
-            dfd.resolve();
-        })
-        .fail(function(ex) {
-            if (ex.problem === "cancelled") {
-                dfd.reject(ex);
-                return;
-            }
-            var error = new Error(buffer.trim());
-            dfd.reject(error);
+    unregisterService.Unregister({})
```

Comment:
Similiar problem to line 144 above.  You'll need to make sure any of the proxies are actually ready before calling any of them just in case the proxy isn't ready yet.